### PR TITLE
USWDS-Site - Dependencies: Remove twig html loader and update build script

### PR DESCRIPTION
**Pull request by mejiaj**
# Summary

Use this to test the feature branch uswds/uswds#6082 that removes the unused twig html loader dependency. It also fixes a build issue by removing the legacy peer dependencies flag in the `build-uswds-if-needed` gulp task.

> [!Important]
> We need to update the uswds branch before merge.

## Preview link

[Preview link → ](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/remove-deps-uswds-6082/)

[Component previews → ](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/remove-deps-uswds-6082/components/overview/)
Make sure that previews and code samples are correct.

Doesn't hurt to check generated pages from USWDS markdown files, like [Security](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/remove-deps-uswds-6082/about/security/).

## Testing and review

Site should build without any issues or regressions.

## Additional context

Renamed branch to reflect that it'll need to be merged and not just for testing.

If you want to rename your local branch follow the commands below.

```
git branch -m test-remove-deps-uswds-6082 remove-deps-uswds-6082
git fetch origin
git branch -u origin/remove-deps-uswds-6082 remove-deps-uswds-6082
git remote set-head origin -a
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +97,6 @@
   "snyk": true,
   "dependencies": {
     "@uswds/compile": "1.2.0",
-    "@uswds/uswds": "3.9.0"
+    "@uswds/uswds": "github:uswds/uswds#jm-remove-deps-6082"
```
Comment (mejiaj) [inline package.json:100]: **chore**: We'll have to update this before merge.

**Review by amyleadem — APPROVED** [review]
LGTM!

- [x] Confirm that removing the `--legacy-peer-deps` allows uswds to build successfully without the `twig-html-loader` dependency.
- [x] Confirm I can run a clean `npm install`, clean `npm run build`, `npm start` and `npm run test` without error.
- [x] Confirm that component previews are rendering as expected.

File: .tool-versions [hunk 2]
```diff
@@ -1,2 +1,2 @@
 ruby 3.2.5
-nodejs 20.9.0
```
Comment (thisisdano) [inline .tool-versions:2]: Updated this to LTS for asdf users

Comment (thisisdano) [reply .tool-versions:2]: (Added here since I needed it to install the develop branch of uswds)

**Pull request merged**